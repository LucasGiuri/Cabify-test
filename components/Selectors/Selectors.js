import React, { Fragment } from 'react';
import Button from '../Button/Button';
import './Selectors.scss';
import PropTypes from 'prop-types';

const Selectors = ({selectors, onChange, onClick}) => {
  return (
    <Fragment>
      {selectors.map(selector => (
        <div className="selectors__selector" key={selector.id}>
          <select className="selectors__selector--select" onChange={() => onChange(selector.id, event.target.value)}>
            <option value=''>Choose a service...</option>
            {selector.services.map(service => (
              <option key={`s-${selector.id}-o-${service.id}`} value={service.value}>
                {service.value}
              </option>
            ))}
          </select>
          <Button text="Remove" classes={['button-danger']} onClick={() => onClick(selector.id)} isDisabled={selectors.length === 1} />
        </div>
      ))}
    </Fragment>
  )
}

Selectors.propTypes = {
  selectors: PropTypes.arrayOf(PropTypes.object).isRequired,
  onChange: PropTypes.func,
  onClick: PropTypes.func
};

Selectors.defaultProps = {
  onChange: () => '',
  onClick: () => ''
};

export default Selectors;