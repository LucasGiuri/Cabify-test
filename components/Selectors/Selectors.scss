.selectors__selector {
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 12px;
}

.selectors__selector--select {
  width: 100%;
  height: 28px;
  margin-right: 16px;
}