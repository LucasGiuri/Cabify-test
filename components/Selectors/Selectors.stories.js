import React from 'react';
import { storiesOf } from '@storybook/react';
import Selectors from './Selectors';

storiesOf('Selectors')
  .add('Selectors default', () => {
    const selectors = [{id: 'FakeSelector1', services: [{id: 'FakeOption1', title: 'FakeOption'}]}];
    
    return <Selectors selectors={selectors} />
  })
