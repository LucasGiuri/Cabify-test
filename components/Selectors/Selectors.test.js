import React from 'react';
import renderer from 'react-test-renderer';
import Selectors from './Selectors';

jest.mock('../Button/Button', () => 'Buttno');

describe('Selectors', () => {
  it('Selectors with required params, basic render snapshoot', () => {
    const selectors = [{id: 'FakeSelector1', services: [{id: 'FakeOption1', title: 'FakeOption'}]}];
    const component = renderer.create(<Selectors selectors={selectors} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Selectors prop onClick is triggered. onClick prop should be called as expected', () => {
    const onClick = jest.fn();
    const selectors = [{id: 'FakeSelector1', services: [{id: 'FakeOption1', title: 'FakeOption'}]}];
    const componentToTest = (
      <Selectors selectors={selectors} onClick={onClick} />
    );
    const tree = renderer.create(componentToTest);
    const buttno = tree.root.findByType("Buttno");
    buttno.props.onClick();
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith("FakeSelector1");
  })
})
