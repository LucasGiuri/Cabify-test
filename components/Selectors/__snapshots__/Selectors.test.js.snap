// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selectors Selectors with required params, basic render snapshoot 1`] = `
<div
  className="selectors__selector"
>
  <select
    className="selectors__selector--select"
    onChange={[Function]}
  >
    <option
      value=""
    >
      Choose a service...
    </option>
    <option />
  </select>
  <Buttno
    classes={
      Array [
        "button-danger",
      ]
    }
    isDisabled={true}
    onClick={[Function]}
    text="Remove"
  />
</div>
`;
